sunpy/physics/differential_rotation.py

```diff
@@ -84,138 +88,318 @@ def diff_rot(duration: u.s, latitude: u.deg, rot_type='howard', frame_time='side
+        the output is specified in the heliographic Stonyhurst coordinate system.
+    """
+    # Check the input and create the new observer
+    if (observer is not None) and (time is not None):
+        raise ValueError("Either the 'observer' or the 'time' keyword must be specified, but not both simultaneously.")
+    elif observer is not None:
+        # Check that the new_observer is specified correctly.
+        if not (isinstance(observer, (BaseCoordinateFrame, SkyCoord))):
+            raise ValueError(
+                "The 'observer' must be an astropy.coordinates.BaseCoordinateFrame or an astropy.coordinates.SkyCoord.")
+        if observer.obstime is None:
+            raise ValueError("The observer 'obstime' property must not be None.")
+        new_observer = observer
+    elif time is not None:
+        warnings.warn("Using 'time' assumes an Earth-based observer.")
+        if isinstance(time, TimeDelta) or isinstance(time, u.Quantity):
+            new_observer_time = initial_obstime + time
+        else:
+            new_observer_time = parse_time(time)
+
+        new_observer = get_body("earth", new_observer_time)
+    else:
+        raise ValueError("Either the 'observer' or the 'time' keyword must not be None.")
+
+    return new_observer.transform_to(HeliographicStonyhurst)
+
+
+def solar_rotate_coordinate(coordinate, observer=None, time=None, **diff_rot_kwargs):
     """
     Given a coordinate on the Sun, calculate where that coordinate maps to
-    at some later or earlier time, given the solar rotation profile.
+    as seen by a new observer at some later or earlier time, given that
+    the input coordinate rotates according to the solar rotation profile.
+
+    The amount of solar rotation is based on the amount of time between the
+    observation time of the input coordinate and the observation time of the
+    new observer. The new observer is specified in one of two ways, either
+    using the "observer" or "time" keywords.
+
+    If the "observer" keyword is set, it is used to specify the location
+    of the new observer in space and time.  The difference between the
+    coordinate time and the new observer time is used to calculate the amount
+    of solar rotation applied, and the location of the new observer in space
+    is used to calculate where the rotated coordinate is as seen from the
+    new observer.
 
-    Note that if the new observer location is defined using a
-    BaseCoordinateFrame or SkyCoord, then it is assumed that the new observer
-    location is correct for the new observer time that was also passed in.
+    If the "time" keyword is set, it is used to specify the number of
+    seconds to rotate the coordinate by. Note that using the "time" keyword
+    assumes that the new observer is on the Earth. This may be a reasonable
+    assumption depending on the application.
+
+    Either the "observer" or "time" keyword must be specified, but both
+    cannot be specified at the same time.
```

Comment:
Same as the other function, these notes could just be in the Parameters section instead.